Content.Server/_Goobstation/Heretic/Ritual/CustomBehavior.Sacrifice.cs

```diff
@@ -87,37 +152,157 @@ public override bool Execute(RitualData args, out string? outstr)
         return true;
     }
 
+
+    
     public override void Finalize(RitualData args)
     {
         for (int i = 0; i < Max; i++)
         {
             var isCommand = args.EntityManager.HasComponent<CommandStaffComponent>(uids[i]);
-            var knowledgeGain = isCommand ? 2f : 1f;
+            var knowledgeGain = isCommand ? 4f : 2f;
 
-            // YES!!! GIB!!!
-            if (args.EntityManager.TryGetComponent<DamageableComponent>(uids[i], out var dmg))
+            // start the sacrifing process -space
+            if (args.EntityManager.TryGetComponent<TransformComponent>(uids[i], out var transform))
             {
-                var prot = (ProtoId<DamageGroupPrototype>) "Brute";
-                var dmgtype = _proto.Index(prot);
-                _damage.TryChangeDamage(uids[i], new DamageSpecifier(dmgtype, 1984f), true);
-            }
 
-            if (args.EntityManager.TryGetComponent<HereticComponent>(args.Performer, out var hereticComp))
-                _heretic.UpdateKnowledge(args.Performer, hereticComp, knowledgeGain);
+                //drop all items -space
 
-            // update objectives
-            if (_mind.TryGetMind(args.Performer, out var mindId, out var mind))
-            {
-                // this is godawful dogshit. but it works :)
-                if (_mind.TryFindObjective((mindId, mind), "HereticSacrificeObjective", out var crewObj)
-                && args.EntityManager.TryGetComponent<HereticSacrificeConditionComponent>(crewObj, out var crewObjComp))
-                    crewObjComp.Sacrificed += 1;
-
-                if (_mind.TryFindObjective((mindId, mind), "HereticSacrificeHeadObjective", out var crewHeadObj)
-                && args.EntityManager.TryGetComponent<HereticSacrificeConditionComponent>(crewHeadObj, out var crewHeadObjComp)
-                && isCommand)
-                    crewHeadObjComp.Sacrificed += 1;
+
+                if (args.EntityManager.TryGetComponent<InventoryComponent>(uids[i], out var comp2))
+                {
+                    var transformSystem = _system.EntityManager.System<TransformSystem>();
+                    var inventorySystem = _system.EntityManager.System<InventorySystem>();
+                    var sharedPopupSystem = _system.EntityManager.System<SharedPopupSystem>();
+
+                    foreach (var item in inventorySystem.GetHandOrInventoryEntities(uids[i]))
+                    {
+                        transformSystem.DropNextTo(item, uids[i]);
+                    }
+                }
+
+
+
+
+                // teleports them away -space
+                var mapPos = _xform.GetWorldPosition(transform);
```

Comment:
Break this functionality up into a separate function, there's way to much going on here